common/djangoapps/student/views.py

```diff
@@ -693,11 +707,13 @@ def dashboard(request):
         'order_history_list': order_history_list,
         'courses_requirements_not_met': courses_requirements_not_met,
         'nav_hidden': True,
+        'course_programs': course_programs,
+        'program_marketing_url': urljoin(settings.MKTG_URLS.get('ROOT'), 'xseries' + '/marketing')
```

Comment:
to clarify: the base url for marketing links is as you have it above, except for the '/marketing' part which should go away.   each time you display a program you append the marketing_slug for that program to the base url, to the get right link.